### PR TITLE
h5z-zfp: Convert to CMakePackage

### PR DESCRIPTION
**Pull request by jrood-nrel**

### CODE REVIEW

File: var/spack/repos/builtin/packages/h5z-zfp/package.py [hunk 1]
```diff
@@ -6,15 +6,15 @@
...
     """A highly flexible floating point and integer compression plugin for the
        HDF5 library using ZFP compression."""
 
     homepage = "https://h5z-zfp.readthedocs.io/en/latest"
     git      = "https://github.com/LLNL/H5Z-ZFP.git"
     url      = "https://github.com/LLNL/H5Z-ZFP/archive/refs/tags/v1.0.1.tar.gz"
 
-    version('develop', branch='master')
+    version('develop', branch='master', preferred=True)
     version('1.0.1', sha256='b9ed91dab8e2ef82dc6706b4242c807fb352875e3b21c217dd00782dd1a22b24')
```
Comment (chuckatkins) [inline var/spack/repos/builtin/packages/h5z-zfp/package.py:18]: The CMake build wasn't added until after the 1.0.1 release so none of the specified versions are even valid for this package.  @brtnfld can we get a new tag or release of the plugin to use for the package so this doesn't have to try to manage both build systems?

File: var/spack/repos/builtin/packages/h5z-zfp/package.py [hunk 2]
```diff
@@ -33,13 +33,31 @@ class H5zZfp(MakefilePackage):
     patch('Makefile.0.7.0.patch', when='@0.7.0')
```
Comment (chuckatkins) [inline var/spack/repos/builtin/packages/h5z-zfp/package.py:33]: The `Makefile` patch is no longer needed now that the package.py is building with cmake

File: var/spack/repos/builtin/packages/h5z-zfp/package.py (hunk 1 above)
Comment (jrood-nrel) [reply var/spack/repos/builtin/packages/h5z-zfp/package.py:18]: It uses the previous build strategy for the existing versions.

**Comment by jrood-nrel** [comment]
Maybe we can just remove the old versions and make this a `CMakePackage` using only the git version for now? I don't think this is a high traffic package.

**Comment by alalazo** [comment]
Seems another case for #30411

**Comment by alalazo** [comment]
@jrood-nrel We have support for multiple build systems since v0.19.0 Do you want to rewrite this package to switch the build system?

**Comment by jrood-nrel** [comment]
Sure. I'm not sue when I will get to it. I will close this until I get to it.

**Pull request closed without merging**